src/components/BoxSelection/BoxSelection.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from 'react';
+import { TPAComponentProps } from '../../types';
+import { st, classes } from './BoxSelection.st.css';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+import { Option as BoxSelectionOption } from './Option';
+
+export enum BoxSize {
+  small = 'small',
+  xLarge = 'large',
+}
+
+export interface BoxSelectionProps extends TPAComponentProps {
+  /**
+   * Is used to group options together
+   */
+  name: string;
+  /**
+   * size variations
+   */
+  size?: BoxSize;
+  /**
+   * Used for SR labelling the group of options
+   */
+  'aria-label'?: string;
+  /**
+   * Used for SR labelling the group of options (by id)
+   */
+  'aria-labelledby'?: string;
+  /**
+   * The onChange callback
+   */
+  onChange?(id: string): void;
+  /**
+   * The Box selected option value
+   */
+  value?: string;
+  /**
+   * The Box selected childrens
+   */
+  children?: React.ReactNode;
+}
+
+interface DefaultProps {
+  size?: BoxSize;
+  'aria-label'?: string;
+  'aria-labelledby'?: string;
+  value: string;
+}
+
+/** The box selection is used to give the user to select single or multiple boxes. */
+export class BoxSelection extends React.Component<BoxSelectionProps> {
+  static displayName = 'BoxSelection';
+  static Option = BoxSelectionOption;
+
+  static defaultProps: DefaultProps = {
+    size: BoxSize.xLarge,
+    'aria-label': '',
+    'aria-labelledby': '',
+    value: '',
+  };
+  // state = {
+  //   value: '',
+  // };
+
+  // _onChange(value: string) {
+  //   this.setState({ value });
+  // }
+
+  render() {
+    const {
+      className,
+      children,
+      size,
+      name,
+      ['aria-label']: ariaLabel,
+      ['aria-labelledby']: ariaLabelledBy,
+    } = this.props;
+
+    //const { value } = this.state;
+    return (
+      <TPAComponentsConsumer>
+        {() => (
+          <div
+            className={st(classes.root, {}, className)}
+            data-hook={this.props['data-hook']}
+            aria-label={ariaLabel}
+            aria-labelledby={ariaLabelledBy}
+          >
+            {React.Children.map(
+              children,
+              (child: BoxSelectionOption, key: number) => {
+                return (
+                  React.isValidElement(child) && (
+                    <>
+                      {React.cloneElement(child, {
+                        key,
+                        id: `"${key}"`,
+                        name,
+                        size,
+                        checked: child.props.id === this.props.value,
+                        className: 'boxSelectionOption',
+                        onChange: (e) => {
+                          this.props.onChange(e.value);
+                          //this._onChange(e.value);
+                        },
```

Comment:
So actually, instead of passing the `onChange` method to the children, we can use event delegation, and add the `onChange` callback directly on the wrapping `div`.

So it would look like this:
```tsx
render() {
  ...
  return (
    <div className={st(classes.root, {}, className)}
            data-hook={this.props['data-hook']}
            aria-label={ariaLabel}
            aria-labelledby={ariaLabelledBy}
            onChange={e => this.props.onChange({id: e.target.getAttribute('data-id')})}
            >
      {React.Children.map(
        ...
      )}
    <div>
  );
}
```
I used the `data-id` attribute to save the `id` on the element itself

Reply:
So I change it as you described and changed it to '**onClick**' event since the div that wraps the children is not triggered on the 'Change' event.

There is still some problem (that you can see on the Settings panel) when the user clicks on the button, it works fine, but when he clicks on the children button (the text), it doesn't receive the data-id, and therefore, there is no change.

So I'm trying to figure out what can we do here. 
If you tackled this before please let me know.

Reply:
So I added to the child to have the 'onChange' event as you suggested in a different comment.